_posts/2019-09-19-sorting-a-table-with-live-view-live-links.md

```diff
@@ -0,0 +1,197 @@
+---
+author: Sophie DeBenedetto
+author_link: https://github.com/sophiedebenedetto
+categories: post
+tags: ['live view']
+date: 2019-09-08
+layout: post
+title: Building a Table Sort UI with Live View's `live_link`
+excerpt: >
+  We'll use LiveView's `live_link/3` together with the `handle_params/3` callback to allow users to sort a table in real-time.
+---
+
+LiveView makes is easy to solve for some of the most common UI challenges with little to no front-end code. It allows us to save JavaScript for the hard stuff––for complex and sophisticated UI changes. In building out a recent admin-facing view that included a table of student cohorts at the Flatiron School, I found myself reaching for LiveView. In just a few lines of backend code, my sortable table was up and running. Keep reading to see how you can leverage LiveView's `live_link/2` and `handle_params/3` to build out such a feature.
+
+## The Feature
+
+Our view presents a table of student cohorts that looks like this:
+
+![live view table]({% asset live-view-table.png @path %})
+
+Users need to be able to sort this table by cohort name, campus, start date or status. We'd also like to ensure that the "sort by" attribute is included in the URL's query params, so that users can share links to sorted views.
+
+Here's a look at the behavior we're going for. Note how the URL changes when we click on a given column heading to sort the table.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/-4VRaX1uEhk" frameborder="0" allow="accelerometer; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+## Using `live_link/3`
+
+LiveView's [`live_link/2`](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#module-live-navigation) function allows page navigation using the browser's [pushState API](https://developer.mozilla.org/en-US/docs/Web/API/History_API). This will ensure that that URL will change to include whatever parameters we include in a given `live_link/2` call.
+
+One important thing to note before we proceed, however. In order to use LiveView's live navigation features, our LiveView needs to be mounted directly in the router, _not_ live rendered from a controller action.
```

Comment:
saying `LiveView` or `live` too many times in this sentence.
Can we simplify this?